backend/fms_core/resources/sample.py

```diff
@@ -177,6 +180,12 @@ def import_obj(self, obj, data, dry_run):
             individual = None
             errors["individual"] = ValidationError(e.messages.pop(), code="invalid")
 
+        if data["Project"]:
+            try:
+                self.project = Project.objects.get(name=get_normalized_str(data, "Project"))
```

Comment:
self.Project went from a Field() to a Dict() to a Project() ... this is not good... You already create the relationship by creating the samplebyproject entry in the m2m section. You should not set self.project here and instead simply test for the project with exists() and raise and error if it does not exist.

Reply:
Agreed. Also, btw if you want to use it for an object, you can initiate it by default with variable_name = None. if it is meant to represent a dictionary you can initiate it with variable_name = {}